enterprise/api/v1/views.py

```diff
@@ -510,3 +515,94 @@ class EnterpriseCustomerReportingConfigurationViewSet(EnterpriseReadOnlyModelVie
     )
     filter_fields = FIELDS
     ordering_fields = FIELDS
+
+
+class CouponCodesView(APIView):
+    """
+    API to request coupon codes.
+    """
+    permission_classes = (permissions.IsAuthenticated,)
```

Comment:
I think we need an extra permission class here to check for the enterprise_api_data_access group to restrict it to only users that have portal access.

I wouldn't want any user to be able to spam CS by hitting this endpoint.